scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift

```diff
@@ -21,34 +21,61 @@ import XCTest
 
 class FirebaseCppUITestAppUITests: XCTestCase {
 
-  override func setUp() {
-    // Put setup code here. This method is called before the invocation of each test method in the class.
-
-    // In UI tests it is usually best to stop immediately when a failure occurs.
-    continueAfterFailure = false
-
-    // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
-    XCUIApplication().launch()
-
-    // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
-  }
+  func testGMA() {
+    // Launch this Helper App
+    let helperApp = XCUIApplication()
+
+    // Periodically check and dismiss dialogs with "Allow" or "OK"
+    Timer.scheduledTimer(withTimeInterval: 5, repeats: true) { (_) in
+#if os(iOS)
+      NSLog("finding springboard ...")
+      let springboard = XCUIApplication(bundleIdentifier: "com.apple.springboard")
+      for button in [springboard.buttons["Open"], springboard.buttons["Allow"], springboard.buttons["OK"]] {
+        if button.exists {
+          NSLog("Dismissing system dialog")
+          button.tap()
+        }
+      }
+#elseif os(tvOS)
+      NSLog("finding pineboard ...")
+      let pineboard = XCUIApplication(bundleIdentifier: "com.apple.PineBoard")
+      for button in [pineboard.buttons["Open"], pineboard.buttons["Allow"], pineboard.buttons["OK"]] {
+        if button.exists {
+          NSLog("Dismissing system dialog")
+          let remote: XCUIRemote = XCUIRemote.shared
+          remote.press(.select)
+        }
+      }
+#endif
+    }
 
-  override func tearDown() {
-    // Put teardown code here. This method is called after the invocation of each test method in the class.
-  }
+    // Launch UI Test App
+    helperApp.launch()
+    // Wait until UI Test App open Integration Test App
+    helperApp.wait(for: .runningBackground, timeout: 20)
+
+    // Wait until Integration Test App closed (testing finished)
+    let expectation = XCTestExpectation(description: "Integration Test App closed")
+    Timer.scheduledTimer(withTimeInterval: 5, repeats: true) { (_) in
+      if helperApp.state == .runningForeground {
+        NSLog("Integration Test App closed... UI Test App back to foreground...")
+        expectation.fulfill()
+      } else {
+        NSLog("Testing... UI Test App in background...")
+      }
+    }
 
 
-  func testGMA() {
+    // Start Automated UI Test
     let app = XCUIApplication(bundleIdentifier: "com.google.ios.admob.testapp")
-    app.launch()
```

Comment:
why is this testapp no longer launched?

Reply:
It's launch here: https://github.com/firebase/firebase-cpp-sdk/blob/0843d25c77babfbd5b31cd48d2933de8f93eb4b7/scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/ViewController.swift#L33